### DISCUSSION TITLE
Play specific files on condition

### DISCUSSION THREAD

**Discussion by kiliw** [pschatzmann/arduino-audio-tools#944] [Q&A] [answered]
Hello everyone,

thanks for the great library, setup was easy and very well documented. I got the _player-sdfat-i2s_ example up and running. My main goal is to have a function (something like `playFile(pathToFile)`) to play a specific file once a criteria is met (like a button press). 
First, I disabled the `AutoNext` and then selected the desired file with `player.setPath("/01/001.mp3")`

```
  player.setPath("/01/001.mp3");
  player.setAutoNext(false);
  
}

void loop() {
  
  player.copy();
  
}
```
Result: I can listen to this file, once. So far, so good.

What I prefer to have is
```
  player.setAutoNext(false);
  
}

void loop() {
if(condition1){
playFile("/01/001.mp3");
}  
if(condition2){
playFile("/01/002.mp3");
}  
... 
}
```

Can someone please point me in the right direction, how to implement this?

I think the `player.copy()` should run in a while loop as long there is data in the stream for the file.

Best
Kilian

**Comment by pschatzmann** [comment, marked as answer]
There is no reason why calling player.setPath() in the loop should not work. 
Just make sure that it is only called once (and not constantly which would disturb the processing).
